src/vm/ast/definition_sorter/mod.rs

```diff
@@ -210,8 +212,12 @@ impl<'a> DefinitionSorter {
                                 }
                             }
                         } else if let Some(native_function) =
-                            NativeFunctions::lookup_by_name(function_name)
-                        {
+                            // NOTE: can use ClarityVersion::latest() here only as long as NO NEW FUNCTIONS are special cased
```

Comment:
Isn't this something that will realistically happen in a hypothetical Clarity3?  Does it make sense to front-load the handling for this case now, while it's still fresh in our minds?

Reply:
I'm not sure -- I think ultimately the way to handle this would be to just create different definition sorters for each Clarity version. We could create a top-level match that then routed the version to the correct definition sorter.